zerver/lib/user_groups.py

```diff
@@ -726,7 +719,17 @@ def set_defaults_for_group_settings(
         else:
             default_group_name = permission_config.default_group_name
 
-        default_group = system_groups_name_dict[default_group_name].usergroup_ptr
+        if default_group_name == "group_creator":
+            if user_group.creator:
+                default_group = UserGroup(
+                    realm=user_group.realm,
+                )
+                default_group.save()
+                default_group.direct_members.set([user_group.creator])
+            else:
+                raise AssertionError("Group creator should not be None.")
```

Comment:
Discussed to raise error at https://chat.zulip.org/#narrow/stream/3-backend/topic/defaults.20other.20than.20system.20groups.20for.20settings/near/1946870